sprint_17

Ramilyushka opened the pull request.

Ramilyushka committed:
-

Ramilyushka committed:
Localization

Ramilyushka committed:
темная тема

Ramilyushka committed:
скриншот тесты

Ramilyushka committed:
import YandexMetrica from Cocoapods

Ramilyushka committed:
PodFile

Ramilyushka committed:
Редактирование трекера
Нужно добавить логику изменения категории

Ramilyushka committed:
Редактирование трекера доделано

Ramilyushka committed:
правкки

Ramilyushka committed:
аналитика часть1

Ramilyushka committed:
Закрепленные

Ramilyushka committed:
Добавлен FiltersViewController

Ramilyushka committed:
Логика фильтрации

Ramilyushka committed:
Сделан экран Статистика

amardanov commented on Tracker/Analytics/AnalyticsService.swift:
```diff
@@ -0,0 +1,25 @@
+import YandexMobileMetrica
+
+final class AnalyticsService {
+    //static let shared = AnalyticsService()
```
Закомментированный код и принты лучше не оставлять в коде приложения. Так код будет чище и ревьюеру будет проще проверять за счет уменьшения количества строк

amardanov commented on TrackerTests/TrackerTests.swift:
```diff
@@ -0,0 +1,22 @@
+final class TrackerTests: XCTestCase {
+    func testTrackersViewControllerLightSnapshot() throws {
+        let vc = TrackersViewController()
+        assertSnapshot(matching: vc, as: .image(traits: .init(userInterfaceStyle: .light)))
```
Лучше всегда писать полное название типа объекта при инициализации, вместо `.init`. Так сборка проекта будет происходить чуть быстрее, а если учитывать, что файлов и инициализаций может быть много, то и прирост может быть существенным. То есть тут будет так:

```
.image(traits: UITraitCollection(userInterfaceStyle: .light)
```

Ramilyushka committed:
Исправлены замечания по ревью:
Критические замечания
+При закреплении трекера происходит дублирование одного трекера. Лучше сделать так, чтобы этого дублирования не было, а закреплялся один экземпляр только
+После выбора фильтра, в котором "ничего не найдено", кнопка фильтров пропадает и обратно вернуть дефолтный или выбрать новый фильтр нельзя
P.S. Кнопка фильтрации отсутствует, если на выбранный день нет трекеров

Ramilyushka committed:
Исправлены замечания по комментариям в гите
